lib/hyrax/transactions/steps/delete_access_control.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require 'dry/monads'
+
+module Hyrax
+  module Transactions
+    module Steps
+      ##
+      # Deletes the Hyrax::AccessControlList for any resource with a `#permission_manager`.
+      # If `#permission_manager` is undefined, succeeds.
+      #
+      # @see https://dry-rb.org/gems/dry-monads/1.0/result/
+      class DeleteAccessControl
+        include Dry::Monads[:result]
+
+        ##
+        # @param [Valkyrie::Resource] obj
+        #
+        # @return [Dry::Monads::Result]
+        def call(obj)
+          return Success(obj) unless obj.respond_to?(:permission_manager)
```

Comment:
i think it's at least theoretically possible for an object to have an ACL, but not a permission manager. the permission manager is mainly a tool for holding permission changes and passing them around with an object.

it might be worth refactoring to use `AccessControlList(obj)`? (though i'm unsure of how well that interface handles the possibility of the ACL not yet existing...)

Reply:
This code mirrors what is in the save_access_control step so maybe this change could be explored and tested as a change across the two steps in a follow up PR?